app.js

```diff
@@ -38,6 +84,77 @@ app.post('/vegetables', helpers.validate, (req, res, next) => {
   res.status(201).json(vegetable)
 })
 
+
+app.get('/fruits', (req, res, next) => {
+  const { fruits } = data
+  const { name } = req.query
+  const fruit = fruits.filter(fruit => fruit.name.includes(name))
+
+  if (!name) {
+    res.json(fruits)
+  } else {
+    res.status(200).json(fruit)
+  }
+
+})
+
+
+app.get('/fruits/:id', (req, res, next) => {
+  const { fruits } = data
+  const { id } = req.params
+  const fruit = fruits.find(fruit => fruit.id === id)
+
+  if (!fruit) {
+    const message = `Could not find fruit with ID of ${id}`
+    next({ status: 404, message })
+  }
+
+  res.json(fruit)
+})
+
+
+app.delete('/fruits/:id', (req, res, next) => {
+  let { fruits } = data
+  const { id } = req.params
+  const fruit = fruits.find(fruit => fruit.id === id);
+
+  if (!fruit) {
+    const message = `Could not find fruit with ID of ${id}`
+    next({ status: 404, message })
+  }
+  else {
+    fruits = fruits.filter(fruit => fruit.id !== id)
```

Comment:
`.filter()` is a non-destructive method -- that means it won't permanently change the value of `fruits`. You should try `.splice()` instead.